packages/frameworks-nuxt/src/runtime/lib/server.ts

```diff
@@ -0,0 +1,88 @@
+import { Auth, setEnvDefaults } from "@auth/core"
+import type { AuthConfig } from "@auth/core"
+import { getRequestFromEvent, getBasePath } from "../utils"
+import { defineEventHandler } from "h3"
+import type { H3Event } from "h3"
+import type { Session } from "@auth/core/types"
+
+/**
+ * Handle auth calls in Nuxt project.
+ *
+ *  @example
+ * ```ts title="server/routes/auth/[...].ts"
+ * import { NuxtAuthHandler } from "#auth"
+ * import GitHub from "@auth/nuxt/providers/github"
+ *
+ * export default defineEventHandler(async (event) => {
+ *    return NuxtAuthHandler({ providers: [GitHub] })
+ * })
+ * ```
+ *
+ * @param config AuthConfig
+ * @returns EventHandler
+ */
+export function NuxtAuthHandler(config: AuthConfig) {
```

Comment:
How do you think if we add a combined export like this:
```ts
export function NuxtAuth(config: AuthConfig) {
  return {
    handlers: // the current code for `NuxtAuthHandler`
    auth: // the current code for `auth` aka `getServerSession`
  }
}
```

Later on in the `server/util/auth` file:
```ts
const { handlers, auth as _auth } = NuxtAuth(config)

export function authHandler() {
  return handlers
}

export async function auth(event: H3Event) {
  return await _auth(event)
}


```
It will look more unified with the NextAuth init:
<img width="799" alt="Screenshot 2024-10-27 at 15 45 24" src="https://github.com/user-attachments/assets/f92de50e-b740-43ca-b337-3344d6964e36">

Reply:
I think that could work. Let me refactor